prov/sockets/src/sock_ep_msg.c

```diff
@@ -244,13 +244,12 @@ static int sock_ep_cm_getname(fid_t fid, void *addr, size_t *addrlen)
 	struct sock_ep *sock_ep = NULL;
 	struct sock_pep *sock_pep = NULL;
 
-	if (*addrlen == 0) {
+	if (*addrlen < sizeof(struct sockaddr_in)) {
 		*addrlen = sizeof(struct sockaddr_in);
 		return -FI_ETOOSMALL;
 	}
 
-	*addrlen = MIN(*addrlen, sizeof(struct sockaddr_in));
-
+	*addrlen = sizeof(struct sockaddr_in);
```

Comment:
You want to keep the MIN type of call.  I would add a new var 'buflen', set buflen = MIN(), and use that in the memcpy routines.  Then always set addrlen = sizeof sockaddr_in.